### PR TITLE
Add product tag reporting to client.

### PR DESCRIPTION
**Pull request by awood**

### CODE REVIEW

**Comment by dgoodwin** [comment]
Small checkstyle error failed the build.

**Comment by dgoodwin** [comment]
I'm seeing some breakage:

$ subscription-manager register --username=admin --password=admin --org=virtwho --force
set([]) is not JSON serializable


$ rpm -qa | grep python-rhsm
python-rhsm-debuginfo-1.13.9-1.fc21.x86_64
python-rhsm-1.13.9-1.fc21.x86_64

$ rpm -qa | grep subscrip
subscription-manager-debuginfo-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-plugin-container-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-plugin-ostree-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-migration-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-gui-1.13.12-1.git.14.9ff2a9c.fc21.x86_64
subscription-manager-firstboot-1.13.12-1.git.14.9ff2a9c.fc21.x86_64

**Comment by dgoodwin** [comment]
Oh I see there's a new python-rhsm PR.  Checking it now.

**Comment by alikins** [comment]
Seems like it would be simpler to let tags be a list, and just uniq it before persisting.

**Comment by dgoodwin** [comment]
Ok due to my mis-tag without your second python-rhsm commit, need to bump the required python-rhsm version one more time here.

Registered fine to a non-updated Candlepin. Registered fine to hosted as well.

**Comment by alikins** [comment]
Could we store the installed product tags as a consumer fact? 
(I guess as a single fact, csv'ed?). Then caching wouldn't need to
change. 

Though it does seem generally useful to have the installed product tags info
per consumer. But it also seems like that would live in ConsumerInstalledProduct.
Does that end up breaking the API?

**Comment by awood** [comment]
@alikins I went with keeping it a set because it made testing easier when I was comparing logically equivalent sets but with different ordering.  I was running into the same issue in the `has_changed` method as well, and it just looked ugly to be casting back and forth from list to set in various places.

**Comment by awood** [comment]
Jenkins will need an updated python-rhsm to build correctly.

**Comment by alikins** [comment]
Would it be worth while to go ahead and track the rest of the particular installed product info while we are adding tags? ie, instead of adding a contentTags param, make it OhYeahOtherProductIdInfo and include things like the product id serial and issuers? So it could be mapped to future per org (or sku/sub/etc) products?

Then server side could aggregate the product tags for a consumer.

**Comment by alikins** [comment]
(On the other hand, may make more sense to pretend that product ids don't exists as a thing, and tracking the date gleaned from them where needed make sense... [like the contentTags, etc]).

**Comment by awood** [comment]
@alikins One of the design goals was to not actually be tied too closely to the idea of a "product".  @dgoodwin suggested this and I actually renamed everything from "product_tags" to "content_tags".  As far as storing the tags as facts, I'm a little hesitant to do that because it seems like we'd just be trying to shoehorn them in.  Facts are almost all immutable or rarely changing attributes specific to machine hardware and configuration.  Placing the content tags in there just seemed *off* to me.

**Comment by alikins** [comment]
I had reservations about requiring a custom json encoder, but given it wouldn't even require a custom encoder, that doesn't bug me now. In practice, that's basically our tiny client side objectmapper ;->

So no reservations from me now.

File: test/fixture.py [hunk 1]
```diff
@@ -61,6 +61,11 @@ def __str__(self):
 
 
 class Matcher(object):
+    @staticmethod
+    def set_eq(first, second):
```
Comment (alikins) [inline test/fixture.py]: set_eq doesn't seem used now.

**Comment by alikins** [comment]
Build-A-Bot: retest this please

**Comment by dgoodwin** [comment]
Legit looking test failures.

**Pull request merged**